SPEC/FILESv2.md

```diff
@@ -0,0 +1,237 @@
+
+#### 1.4 Remove `wrap-with-directory` option
+
+Every import will effectively be "wrapped" in a directory so this option is no longer required.
+
+### 2. Rename `get` to `export`
+
+Renaming to `export` will more accurately describe the intention of the method.
+
+### 3. Remove `cat`
+
+For the people to whom cat means 🐈, the API method is removed and `read` used instead, which is the more obvious opposite to `write` anyway and supports IPFS paths.
+
+### 4. Hoist all methods in the `files` namespace to the root level
+
+Methods that are integral for interacting with the InterPlanetary File System will reside on the root namespace. The reasoning is that these commands are important and will be used often so need to be given priority and ease of access over other APIs that IPFS exposes. It will more effectively advertise what the IPFS core functionality is to aid onboarding and understanding of IPFS in general.
+
+For clarity, the API movement/renaming changes are as follows:
+
+| Old name | New name |
+|---|---|
+| `ipfs add` | `ipfs import` |
+| `ipfs cat` | (removed) |
+| `ipfs get` | `ipfs export` |
+| `ipfs ls` | (removed) |
+| `ipfs files cp` | `ipfs cp` |
+| `ipfs files flush` | `ipfs flush` |
+| `ipfs files ls` | `ipfs ls` |
+| `ipfs files mkdir` | `ipfs mkdir` |
+| `ipfs files mv` | `ipfs mv` |
+| `ipfs files read` | `ipfs read` |
+| `ipfs files rm` | `ipfs rm` |
+| `ipfs files stat` | `ipfs stat` |
+| `ipfs files write` | `ipfs write` |
+
+### 5. Allow both IPFS and MFS paths in API methods
+
+Rather than segregating MFS from the rest of IPFS, we can use MFS paths to refer to content on our local node and IPFS paths to refer to content on the wider IPFS network. We can draw an analogy here with way we use Unix paths and URLs today for working with files in our OS and the Internet respectively. This is natural for many computer users and it's widely understood that the path at the end of a domain navigates through a set of files in much the same way as a path navigates through a set of files on an OS.
+
+Where it makes sense, we can allow MFS paths in the root level API methods and IPFS paths in the MFS API methods. This has already been proven possible as many MFS API methods already accept IPFS paths.
+
+| Method | Accepts IPFS paths | Accepts MFS paths |
+|---|---|---|
+| `ipfs cp` | ✅ | ✅ |
+| `ipfs export` | ✅ | ✅ |
+| `ipfs flush` | ❌ | ✅ |
+| `ipfs import` | ❌ | ❌ |
+| `ipfs ls` | ✅ | ✅ |
+| `ipfs mkdir` | ❌ | ✅ |
+| `ipfs mv` | ✅ | ✅ |
+| `ipfs read` | ✅ | ✅ |
+| `ipfs rm` | ❌ | ✅ |
+| `ipfs stat` | ✅ | ✅ |
+| `ipfs write` | ❌ | ✅ |
+
+The `/ipfs` directory in MFS problem can simply be avoided by either assuming IPFS path (the current solution) or by denying writes to a directory of this name.
```

Comment:
If we're going to join `mfs` with everything else, I'd like to namespace MFS under something like `/local` or `/files. That way, we can continue adding new namespaces without clobbering user-defined directories.

* /ipfs
* /ipns
* /ipld?
* /p2p?? (might be nice to expose this as sockets when mounted with fuse).
* /git/Cid/... ???